hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/MessageSendingResult.java

```diff
@@ -56,14 +62,26 @@ public boolean isClientError() {
         return isInFamily(CLIENT_ERROR);
     }
 
+    public boolean isRetryLater() {
+        return statusCode == SERVICE_UNAVAILABLE.getStatusCode();
+    }
+
     private void initializeForStatusCode(int statusCode) {
         this.statusCode = statusCode;
         responseFamily = familyOf(statusCode);
-        if (!isInFamily(SUCCESSFUL)) {
+        if (this.failure == null && !isInFamily(SUCCESSFUL)) {
             this.failure = new InternalProcessingException("Message sending failed with status code:" + statusCode);
         }
     }
 
+    private long extractRetryAfterMillis(Result result) {
+        HttpFields headers = result.getResponse().getHeaders();
+        if (headers.contains(HttpHeader.RETRY_AFTER)) {
+            return Long.parseLong(headers.get(HttpHeader.RETRY_AFTER)) * 1000L;
+        }
+        return 0L;
```

Comment:
I think that better idea would be to retry after backoff from subscription policy rather than hammer again service that is currently not working...